conans/client/rest/rest_client_common.py

```diff
@@ -193,26 +242,10 @@ def search(self, pattern=None, ignorecase=True):
         return [ConanFileReference.loads(reference) for reference in response]
 
     def search_packages(self, ref, query):
-
-        if not query:
-            url = self.router.search_packages(ref)
-            package_infos = self.get_json(url)
-            return package_infos
-
-        # Read capabilities
-        try:
-            _, _, capabilities = self.server_info()
-        except NotFoundException:
-            capabilities = []
-
-        if COMPLEX_SEARCH_CAPABILITY in capabilities:
-            url = self.router.search_packages(ref, query)
-            package_infos = self.get_json(url)
-            return package_infos
-        else:
-            url = self.router.search_packages(ref)
-            package_infos = self.get_json(url)
-            return filter_packages(query, package_infos)
+        """Client is filtering by the query"""
```

Comment:
Capabilities logic moved to `rest_client`. Simplified code, the only "change" is that the query as argument `?query` is now always sent but our server (the one is not capable of complex search) will ignore it.

Reply:
This is finally not strictly necessary, only a refactor. If you want I can extract it for later.